src/main/scala/mesosphere/marathon/core/task/TaskStateChange.scala

```diff
@@ -0,0 +1,27 @@
+package mesosphere.marathon.core.task
+
+import mesosphere.marathon.core.instance.Instance
+
+sealed trait TaskStateChange
+
+object TaskStateChange {
+  case class Update(newState: Task, oldState: Option[Task]) extends TaskStateChange
+  case class Expunge(task: Task) extends TaskStateChange
+  case class NoChange(taskId: Instance.Id) extends TaskStateChange
+  case class Failure(cause: Throwable) extends TaskStateChange
+  object Failure {
+    def apply(message: String): Failure = Failure(TaskStateChangeException(message))
+  }
+}
```

Comment:
Unsure of our team convention. I extracted this code, nearly verbatim, from
another file in the codebase. I'm happy to factor further if it doesn't
meet our team standards.

On Wed, Sep 7, 2016 at 4:50 AM, Karsten Jeschkies <notifications@github.com>
wrote:

> In src/main/scala/mesosphere/marathon/core/task/TaskStateChange.scala
> <https://github.com/mesosphere/marathon/pull/4313#discussion_r77782605>:
>
> > @@ -0,0 +1,27 @@
> > +package mesosphere.marathon.core.task
> > +
> > +import mesosphere.marathon.core.instance.Instance
> > +
> > +sealed trait TaskStateChange
> > +
> > +object TaskStateChange {
> > +  case class Update(newState: Task, oldState: Option[Task]) extends TaskStateChange
> > +  case class Expunge(task: Task) extends TaskStateChange
> > +  case class NoChange(taskId: Instance.Id) extends TaskStateChange
> > +  case class Failure(cause: Throwable) extends TaskStateChange
> > +  object Failure {
> > +    def apply(message: String): Failure = Failure(TaskStateChangeException(message))
> > +  }
> > +}
>
> Is there some convention of object over package? In this case it seems
> that TaskStateChange is rather a package than an object. Just a little
> nitpicking on my side ;)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4313/files/af2ec891fd251891a78b95196c6ef4479487e23d#r77782605>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACPVLHE4e0ceY4st3PFc4eM1zCO5-yZjks5qnnregaJpZM4J2U1J>
> .
>